src/signalcorr.jl

```diff
@@ -64,15 +63,15 @@ function autocov!(
     r::AbstractVector,
     x::AbstractVector,
     lags::AbstractVector{<:Integer},
-    z=zeros(typeof(zero(eltype(x)) / 1), length(x));
+    z=Vector{typeof(zero(eltype(x)) / 1)}(undef, size(x, 1));
     demean::Bool=true
     )
     lx = length(x)
     m = length(lags)
     length(r) == m || throw(DimensionMismatch())
     check_lags(lx, lags)
-    T = typeof(zero(eltype(x)) / 1)
-    demean  ? z = x :   z .= x .- mean(x)
+    z .= x
+    demean && z .= z .- mean(z)
```

Comment:
Simpler and only writing to `z` once and without broadcasting overhead in the second branch:
```suggestion
    demean ? z .= x .- mean(x) : copyto!(z, x) 
```